src/app/src/pages/eval/components/EvaluationPanel.tsx

```diff
@@ -165,6 +170,58 @@ function AssertionResults({ gradingResults }: { gradingResults?: GradingResult[]
   );
 }
 
+function formatGradingPrompt(prompt: string): string {
+  try {
+    const parsed = JSON.parse(prompt);
+    return JSON.stringify(parsed, null, 2);
+  } catch {
+    return prompt;
+  }
+}
+
+function GradingPromptSection({ gradingResults }: { gradingResults?: GradingResult[] }) {
+  const promptsWithData = gradingResults?.filter((r) => r.metadata?.renderedGradingPrompt) || [];
+
+  if (promptsWithData.length === 0) {
+    return null;
+  }
```

Comment:
nit: I think rather than hiding the section, we should maybe write "N/A"? for consistency